hudi-common/src/main/java/org/apache/hudi/common/config/HoodieMetadataConfig.java

```diff
@@ -527,6 +527,33 @@ public final class HoodieMetadataConfig extends HoodieConfig {
       .withDocumentation("Max size in MB below which metadata file (HFile) will be downloaded "
           + "and cached entirely for the HFileReader.");
 
+  public static final ConfigProperty<Boolean> METADATA_HFILE_BLOCK_CACHE_ENABLED = ConfigProperty
+      .key(METADATA_PREFIX + ".hfile.block.cache.enabled")
+      .defaultValue(false)
+      .markAdvanced()
+      .sinceVersion("1.1.0")
+      .withDocumentation("Enable HFile block-level caching for metadata files. This caches frequently "
+          + "accessed HFile blocks in memory to reduce I/O operations during metadata queries. "
+          + "Improves performance for workloads with repeated metadata access patterns.");
+
+  public static final ConfigProperty<Integer> METADATA_HFILE_BLOCK_CACHE_SIZE = ConfigProperty
+      .key(METADATA_PREFIX + ".hfile.block.cache.size")
+      .defaultValue(100)
+      .markAdvanced()
+      .sinceVersion("1.1.0")
+      .withDocumentation("Maximum number of HFile blocks to cache in memory per metadata file reader. "
+          + "Higher values improve cache hit rates but consume more memory. "
+          + "Only effective when hfile.block.cache.enabled is true.");
+
+  public static final ConfigProperty<Integer> METADATA_HFILE_BLOCK_CACHE_TTL_MINUTES = ConfigProperty
+      .key(METADATA_PREFIX + ".hfile.block.cache.ttl.minutes")
+      .defaultValue(60)
+      .markAdvanced()
+      .sinceVersion("1.1.0")
+      .withDocumentation("Time-to-live (TTL) in minutes for cached HFile blocks. Blocks are evicted "
```

Comment:
Does this account for the access pattern, i.e., if the same HFile block is access in an hour from t, t+1min, t+2min, ..., to t+59min, is the block going to be evicted from the block cache at t+60min, or it is evicted at t+119min?

Reply:
Changing this to `expireAfterAccess`.

Good catch! 👍